vector/src/main/java/org/apache/arrow/vector/util/VectorAppender.java

```diff
@@ -698,4 +703,70 @@ public ValueVector visit(ExtensionTypeVector<?> deltaVector, Void value) {
     deltaVector.getUnderlyingVector().accept(underlyingAppender, null);
     return targetVector;
   }
+
+  @Override
+  public ValueVector visit(RunEndEncodedVector deltaVector, Void value) {
+    Preconditions.checkArgument(
+        typeVisitor.equals(deltaVector),
+        "The vector to append must have the same type as the targetVector being appended");
```

Comment:
```suggestion
        "The vector to append must have the same type as the deltaVector being appended");
```

Reply:
Hmm. I think the wording in this message is just terrible, on second glance. We don't have to change them all, but I think the new one should say "The deltaVector to append must have the same type as the targetVector"